### ISSUE TITLE
Excess clipping of nested SVG

### ISSUE THREAD

**Issue by edemaine** [yWorks/svg2pdf.js#244]
**Describe the bug**
Rendering this SVG with a nested `<svg>` tag ends up with a weird clipping box.

**To Reproduce**
Put the following into a `.html` file, then open in a web browser. (Warning: It will immediately trigger a download of file `test.pdf`)

```html
<html>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="148.26817321777344 216.94358825683594 19.859024047851562 24.615386962890625" width="19.859024047851562px" height="24.615386962890625px">
<g transform="translate(147.1999969482422,236.63589477539062)"><g transform="translate(0.5961542100601207 -9.72823342590332) scale(19)"><svg width="1.038678" height="0.536075" viewBox="0 -442 878 453"><defs><path id="MJX-1-TEX-I-1D45A" d="M21 287Q22 293 24 303T36 341T56 388T88 425T132 442T175 435T205 417T221 395T229 376L231 369Q231 367 232 367L243 378Q303 442 384 442Q401 442 415 440T441 433T460 423T475 411T485 398T493 385T497 373T500 364T502 357L510 367Q573 442 659 442Q713 442 746 415T780 336Q780 285 742 178T704 50Q705 36 709 31T724 26Q752 26 776 56T815 138Q818 149 821 151T837 153Q857 153 857 145Q857 144 853 130Q845 101 831 73T785 17T716 -10Q669 -10 648 17T627 73Q627 92 663 193T700 345Q700 404 656 404H651Q565 404 506 303L499 291L466 157Q433 26 428 16Q415 -11 385 -11Q372 -11 364 -4T353 8T350 18Q350 29 384 161L420 307Q423 322 423 345Q423 404 379 404H374Q288 404 229 303L222 291L189 157Q156 26 151 16Q138 -11 108 -11Q95 -11 87 -5T76 7T74 17Q74 30 112 181Q151 335 151 342Q154 357 154 369Q154 405 129 405Q107 405 92 377T69 316T57 280Q55 278 41 278H27Q21 284 21 287Z"/></defs><g transform="scale(1,-1)"><use xlink:href="#MJX-1-TEX-I-1D45A"/></g></svg></g></g>
</svg>
<script type="module">
const {jsPDF} = await import("https://cdn.skypack.dev/pin/jspdf@v2.5.0-qOUBjLwjpjtUHvljumfc/dist=es2020,mode=imports,min/optimized/jspdf.js")
const {svg2pdf} = await import("https://cdn.skypack.dev/pin/svg2pdf.js@v2.2.1-E39EA3q6JjbAU0rSktv0/mode=imports,min/optimized/svg2pdfjs.js")
const width = 19.859024047851562
const height = 24.615386962890625
const pdf = new jsPDF({format: [width, height], orientation: 'portrait'})
await svg2pdf(document.querySelector('svg'), pdf, {width, height})
pdf.save('test.pdf')
</script>
</html>
```

**Expected behavior**

The example [works in the playground](http://raw.githack.com/yWorks/svg2pdf.js/master/index.html?svg=%253Csvg%2520height%253D%252224.615386962890625px%2522%2520width%253D%252219.859024047851562px%2522%2520viewBox%253D%2522148.26817321777344%2520216.94358825683594%252019.859024047851562%252024.615386962890625%2522%2520xmlns%253Axlink%253D%2522http%253A%252F%252Fwww.w3.org%252F1999%252Fxlink%2522%2520xmlns%253D%2522http%253A%252F%252Fwww.w3.org%252F2000%252Fsvg%2522%253E%250A%253Cg%2520transform%253D%2522translate%28147.1999969482422%252C236.63589477539062%29%2522%253E%253Cg%2520transform%253D%2522translate%280.5961542100601207%2520-9.72823342590332%29%2520scale%2819%29%2522%253E%253Csvg%2520viewBox%253D%25220%2520-442%2520878%2520453%2522%2520height%253D%25220.536075%2522%2520width%253D%25221.038678%2522%253E%253Cdefs%253E%253Cpath%2520d%253D%2522M21%2520287Q22%2520293%252024%2520303T36%2520341T56%2520388T88%2520425T132%2520442T175%2520435T205%2520417T221%2520395T229%2520376L231%2520369Q231%2520367%2520232%2520367L243%2520378Q303%2520442%2520384%2520442Q401%2520442%2520415%2520440T441%2520433T460%2520423T475%2520411T485%2520398T493%2520385T497%2520373T500%2520364T502%2520357L510%2520367Q573%2520442%2520659%2520442Q713%2520442%2520746%2520415T780%2520336Q780%2520285%2520742%2520178T704%252050Q705%252036%2520709%252031T724%252026Q752%252026%2520776%252056T815%2520138Q818%2520149%2520821%2520151T837%2520153Q857%2520153%2520857%2520145Q857%2520144%2520853%2520130Q845%2520101%2520831%252073T785%252017T716%2520-10Q669%2520-10%2520648%252017T627%252073Q627%252092%2520663%2520193T700%2520345Q700%2520404%2520656%2520404H651Q565%2520404%2520506%2520303L499%2520291L466%2520157Q433%252026%2520428%252016Q415%2520-11%2520385%2520-11Q372%2520-11%2520364%2520-4T353%25208T350%252018Q350%252029%2520384%2520161L420%2520307Q423%2520322%2520423%2520345Q423%2520404%2520379%2520404H374Q288%2520404%2520229%2520303L222%2520291L189%2520157Q156%252026%2520151%252016Q138%2520-11%2520108%2520-11Q95%2520-11%252087%2520-5T76%25207T74%252017Q74%252030%2520112%2520181Q151%2520335%2520151%2520342Q154%2520357%2520154%2520369Q154%2520405%2520129%2520405Q107%2520405%252092%2520377T69%2520316T57%2520280Q55%2520278%252041%2520278H27Q21%2520284%252021%2520287Z%2522%2520id%253D%2522MJX-1-TEX-I-1D45A%2522%253E%253C%252Fpath%253E%253C%252Fdefs%253E%253Cg%2520transform%253D%2522scale%281%252C-1%29%2522%253E%253Cuse%2520xlink%253Ahref%253D%2522%2523MJX-1-TEX-I-1D45A%2522%253E%253C%252Fuse%253E%253C%252Fg%253E%253C%252Fsvg%253E%253C%252Fg%253E%253C%252Fg%253E%250A%253C%252Fsvg%253E). It looks like this:

![image](https://user-images.githubusercontent.com/2218736/208175371-931c5afa-43cd-461a-89bf-ae5badbd860a.png)

Is there something not yet released to NPM?

**Screenshots**
Instead the path is mostly clipped.  Here are some Illustrator screenshots:

![image](https://user-images.githubusercontent.com/2218736/208176115-f3b75f15-438c-421b-885e-966ef8508304.png)

![image](https://user-images.githubusercontent.com/2218736/208176141-a7bfd9e7-b77a-4aef-8302-69af1c3957db.png)

I'm not really sure where that rectangular bounding box is coming from. Perhaps a `transform` issue?

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Chrome
 - Version: 108.0.5359.100

**Comment by edemaine** [comment]
I did not see any discrepancy between Chrome, Illustrator, or Acrobat in this test case. The playground PDF works everywhere, while the buggy PDF works nowhere.

**Comment by yGuy** [comment]
> The playground PDF works everywhere, while the buggy PDF works nowhere.

Sorry, I cannot reproduce, for me the PDF viewer in the playground shows the pdf, properly, this is in the current version of Edge. Also opening the PDF from the playground in Edge, again, properly shows the PDF. So the PDF viewers seem to disagree, but I take it at least the Adobe tools show the cropping behavior? What version of Chrome are you seeing the clipping behavior in?

**Comment by edemaine** [comment]
The example works fine in the playground; I agree.  What doesn't work is if you use the reproduction above with the .html file.

**Comment by yGuy** [comment]
Interesting. The playground (for some reason, that's not ideal, actually) uses version 2.0.0 of JsPDF which is a lot older than the version you are using. Maybe this is a regression? The svg2pdf version is a slightly newer, unreleased version, but should only differ in third party dependencies during build time. 

Could you try your sampel with 2.0.0 and then maybe bisecting the versions to find the version that changes the behavior?

**Comment by HackbrettXXX** [comment]
I dug into this and it turns out:
* the example works in the playground and does not with the provided JS code, because the playground passes `unit: 'pt'` to the jsPDF constructor.
* Passing 'pt' as unit has the effect that jsPDF's internal scaling factor is `1`. If the option is omitted, the default unit is 'mm', resulting in a scaling factor of `72 / 25.4`. As a consequence, the bounding box of the PDF form object (created by the `<use>`-tag) is wrong. This is definitively a bug in jsPDF:

https://github.com/parallax/jsPDF/blob/5d09af9135a2fe049c7d3c8b95df280d22e4a6db/src/jspdf.js#L5957-L5964

called by 

https://github.com/parallax/jsPDF/blob/5d09af9135a2fe049c7d3c8b95df280d22e4a6db/src/jspdf.js#L5707-L5721

called by 

https://github.com/parallax/jsPDF/blob/5d09af9135a2fe049c7d3c8b95df280d22e4a6db/src/jspdf.js#L5751-L5771

The width and height of the passed bounding box are scaled by the inverse scaling factor and x and y are not.
However, the correct behavior should be that x, y, width, height should all be scaled by the normal (not inverse) scaling factor, so the values passed to `beginFormObject` are interpreted with the correct unit.

To summarize:
* Pass `unit: 'pt'` to the jsPDF constructor as workaround. This should not make a big difference, if you're not using the jsPDF API for other things as svg2pdf.
* I've created an issue on jsPDF side: https://github.com/parallax/jsPDF/issues/3627

**Comment by HackbrettXXX** [comment]
Similar issue as #95.

**Comment by edemaine** [comment]
Awesome, thanks @HackbrettXXX! Adding `unit: 'pt'` fixed this issue and #245.

Closing in favor of https://github.com/parallax/jsPDF/issues/3627
